cmd/util/ledger/migrations/filter_unreferenced_slabs_migration.go

```diff
@@ -0,0 +1,138 @@
+package migrations
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/onflow/atree"
+	"github.com/onflow/cadence/runtime"
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/cmd/util/ledger/reporters"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/convert"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+func StorageIDFromRegisterID(registerID flow.RegisterID) atree.StorageID {
+	storageID := atree.StorageID{
+		Address: atree.Address([]byte(registerID.Owner)),
+	}
+	copy(storageID.Index[:], registerID.Key[1:])
+	return storageID
+}
+
+type FilterUnreferencedSlabsMigration struct {
+	log zerolog.Logger
+	rw  reporters.ReportWriter
+}
+
+var _ AccountBasedMigration = &FilterUnreferencedSlabsMigration{}
+
+const filterUnreferencedSlabsName = "filter-unreferenced-slabs"
+
+func NewFilterUnreferencedSlabsMigration(rwf reporters.ReportWriterFactory) *FilterUnreferencedSlabsMigration {
+	return &FilterUnreferencedSlabsMigration{
+		rw: rwf.ReportWriter(filterUnreferencedSlabsName),
+	}
+}
+
+func (m *FilterUnreferencedSlabsMigration) InitMigration(
+	log zerolog.Logger,
+	_ []*ledger.Payload,
+	_ int,
+) error {
+	m.log = log.
+		With().
+		Str("migration", filterUnreferencedSlabsName).
+		Logger()
+
+	return nil
+}
+
+func (m *FilterUnreferencedSlabsMigration) MigrateAccount(
+	_ context.Context,
+	address common.Address,
+	oldPayloads []*ledger.Payload,
+) ([]*ledger.Payload, error) {
+
+	migrationRuntime, err := newMigratorRuntime(address, oldPayloads)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create migrator runtime: %w", err)
+	}
+
+	storage := migrationRuntime.Storage
+
+	newPayloads := oldPayloads
+
+	err = checkStorageHealth(address, storage, oldPayloads)
+	if err != nil {
```

Comment:
invert condition to reduce nesting.

Reply:
Updated in 5c4feffa909b9158654bc4f754a65c7f08650c33. Left a comment why we're still returning no error at the end